File: src/main/java/com/avant/auth/UserRepository.java
package com.avant.auth;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.mongodb.repository.Query;

public interface UserRepository extends MongoRepository<User, String> {

	@Query ("{'mail' : ?0, 'pass' : ?1}")
	User findByCredAndPass (String cred, String pass);
	
}

File: src/main/java/com/avant/auth/UserRepository.kt
package com.avant.auth

import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.mongodb.repository.Query

interface UserRepository : MongoRepository<User, String> {
	
	@Query("{'mail' : ?0, 'pass' : ?1}")
	fun findByCredAndPass(cred: String, pass: String): User
	
}
